teuthology/task/internal/__init__.py

```diff
@@ -318,7 +318,21 @@ def fetch_binaries_for_coredumps(path, remote):
             # 1422917770.7450.core: ELF 64-bit LSB core file x86-64, version 1 (SYSV), SVR4-style, \
             # from 'radosgw --rgw-socket-path /home/ubuntu/cephtest/apache/tmp.client.0/fastcgi_soc'
             log.info(f' core looks like: {dump_out}')
-            dump_program = dump_out.split("from '")[1].split(' ')[0]
+            if 'gzip' in dump_out:
+                try:
+                    log.info("core is compressed, try accessing gzip file ...")
+                    with gzip.open(dump_path, 'r') as f:
+                        file_content = f.read().decode()
+                    log.info(f' file content looks like: {file_content}')
+                    dump_out = file_content
```

Comment:
`dump_out` is supposed to be the result of running `file` on a core, right?  `dump_program` is then parsed out from that.

However, here, if the core is compressed, `dump_out` gets the decompressed core contents...  If you want to do it this way, you need to run `file` twice.

Reply:
Actually, `file` seems to be able to handle compression natively:
```
     -z, --uncompress
             Try to look inside compressed files.

     -Z, --uncompress-noreport
             Try to look inside compressed files, but report information about the contents only not the
             compression.
```
So `file -z -Z` should do the trick and require no new code.

Reply:
thank you @idryomov for your comments,
So it tested out `file -z -Z` and I think it doesn't really give you the content of what is inside the compressed file, rather it tells you what kind of text is inside the file, usually it is `ASCII text`

Reply:
actually, nvm I think I wasn't familiar with how core formats are when you use `file` on it. Now I get it, thank you.